pdns/recursordist/rec-tcpout.hh

```diff
@@ -48,27 +48,30 @@ public:
     }
 
     std::shared_ptr<TCPIOHandler> d_handler;
+    std::optional<ComboAddress> d_local;
     timeval d_last_used{0, 0};
     size_t d_numqueries{0};
   };
 
-  void store(const struct timeval& now, const ComboAddress& ip, Connection&& connection);
-  Connection get(const ComboAddress& ip);
+  using pair_t = std::pair<ComboAddress, std::optional<ComboAddress>>;
```

Comment:
here I realise you already have a nice typedef. Now to think of a better name. RemoteWithOptionalLocal is a bit long :)